### PR TITLE
Updating the test status to POSITIVE/NEGATIVE

### LINKED ISSUE #1167: Error in Updating test results
error 2

When you update a patient's test result as 'awaiting' and then save it, there isn't a way to update it as 'positive' or 'negative' later on.

### PR DESCRIPTION
**Pull request by Sandra3657**
Resolves #1167

In the Sample Test History for a patient - If the current status and result is COMPLETED and AWAITING respectively, it is now possible to edit the result to POSITIVE or NEGATIVE.

Anim

### CODE REVIEW

**Comment by amruthavijay88** [comment]
Tested - When we select 'test completed' and click 'awaiting' - there is no option to update it as positive/negative later.

File: src/Components/Patient/UpdateStatusDialog.tsx [hunk 1]
```diff
@@ -1,162 +1,193 @@
...
+
+  const currentStatus = SAMPLE_TEST_STATUS.find(
+    (i) => i.text === sample.status
+  );
+
+  const status = String(sample.status) as keyof typeof SAMPLE_FLOW_RULES;
+  const validStatusChoices = statusChoices.filter(
+    (i) => status && statusFlow[status] && statusFlow[status].includes(i.text)
+  );
+  // .filter(i => roleStatusMap[userType] && roleStatusMap[userType].includes(i.text))
```
Comment (smartclash) [inline src/Components/Patient/UpdateStatusDialog.tsx:83]: can we remove this?

**Review by smartclash — CHANGES_REQUESTED** [review]

**Comment by Sandra3657** [comment]
Can you give the steps to recreate the issue? @amruthavijay88

**Comment by amruthavijay88** [comment]
@Sandra3657 - we had discussed on steps over yesterday's EOD call? Please confirm if you need clarity.

**Comment by Sandra3657** [comment]
Yes please. I have tried recreating the steps, and their was no issue. So I may have done it wrong. So, can you please repeat the steps? @amruthavijay88.

**Comment by amruthavijay88** [comment]
1. Open a patient card
2.  Click Request Sample test
3.  Fill in details and assign test facility name 'hi tech test' and submit
4.  Logout and login as user of Hi tech test facility
5. Click sample test
6. Go until test completed and click awaiting
7.  Card freezes - you can't made further edits if you put awaiting.

**Comment by amruthavijay88** [comment]
@Sandra3657

**Comment by aparnacoronasafe** [comment]
tested in preview

**Comment by amruthavijay88** [comment]
Tested - failed. Screenshots below - 

1. While adding 'awaiting status'
image

2. While trying to update status as positive/negative - is taken off.

image

**Comment by Sandra3657** [comment]
It is working in preview. @amruthavijay88 

test

**Comment by nihal467** [comment]
error 1

step 1: Update the sample test  to 'approved'
step 2: Select 'new status', choose 'test completed' and in Result choose 'invalid'

Even the test in completed, the 'update sample test status' button appear in the screen

**Comment by nihal467** [comment]
test approved

**Comment by bodhish** [comment]
@Sandra3657 can you fix the merge conflicts

**Pull request merged**